spec/webauthn/authenticator_data/attested_credential_data_spec.rb

```diff
@@ -6,6 +6,9 @@ def raw_attested_credential_data(options = {})
       aaguid: SecureRandom.random_bytes(16),
       id: SecureRandom.random_bytes(16),
       public_key: CBOR.encode(
+        1 => 2,
```

Comment:
Not sure if it's worth it, but would be a bit more easy to understand the specs if you use the constants from the COSE::Key modules in the same way you did it for https://github.com/cedarcode/webauthn-ruby/blob/cb40f68b49c40b9802ffdfc80d300b23cb4d5e3d/spec/support/fake_authenticator.rb#L102

Reply:
I agree with you in that this test case setup is very difficult to understand for any reader, and it needs a change. Good call.

The reason I am not sure about using the internal implementation constants in the test case setup is that the test looses the characteristic of being decoupled from the implementation.

What I mean is:

Case 1:
* Given that we are using `1 => 2` in the test
* And that someone in the future accidentally changes the `KTY_EC2` constant from `2` to `3`
* Then the test fails

Case 2:
* Given that we are using `1 => COSE::Key::EC2::KTY_EC2`
* And that someone in the future accidentally changes the `KTY_EC2` key from `2` to `3`
* Then the test is still green

I do value the fact that the test would fail in that case, and would make us realize that doing that would break what's expected from the spec.

An approach that I can think of would improve code readability that still retains decoupling could be doing something like:

```diff
diff --git a/spec/webauthn/authenticator_data/attested_credential_data_spec.rb b/spec/webauthn/authenticator_data/attested_credential_data_spec.rb
index 9c0152b..b122622 100644
--- a/spec/webauthn/authenticator_data/attested_credential_data_spec.rb
+++ b/spec/webauthn/authenticator_data/attested_credential_data_spec.rb
@@ -2,11 +2,14 @@
 
 RSpec.describe WebAuthn::AuthenticatorData::AttestedCredentialData do
   def raw_attested_credential_data(options = {})
+    kty_label = 1
+    kty_ec2 = 2
+
     options = {
       aaguid: SecureRandom.random_bytes(16),
       id: SecureRandom.random_bytes(16),
       public_key: CBOR.encode(
-        1 => 2,
+        kty_label => kty_ec2,
         3 => -7,
         -1 => 1,
         -2 => SecureRandom.random_bytes(32),
```

Thoughts?

Reply:
Cool.

Just realized line 102 you referred to was in the FakeAuthenticator, which is test setup. I should probably decouple that as part of this...